cognite/client/_cognite_client.py

```diff
@@ -30,6 +30,8 @@ class CogniteClient:
         max_workers (int): Max number of workers to spawn when parallelizing data fetching. Defaults to 10.
         headers (Dict): Additional headers to add to all requests.
         timeout (int): Timeout on requests sent to the api. Defaults to 30 seconds.
+        token_factory(Callable[[], str]): A method which takes no arguments and returns a token to use for authentication.
+            This will override any api-key set.
```

Comment:
Would we reduce confusion and potential for bugs if we enforce mutual exclusivity instead (throwing an exception if both are set)?

Reply:
I think a single parameter is more pythonesque, and fine if well documented.

Reply:
I prefer it overriding. That way I won't need to unset the COGNITE_API_KEY environment variable each time I want to use tokens.

Reply:
@sanderland I think he was referring to api-key/token not token_factory/token

Reply:
ah, ignore the above then :)

Reply:
test for string option?